packages/nuxt/src/app/nuxt.ts

```diff
@@ -425,6 +425,11 @@ export async function applyPlugins (nuxtApp: NuxtApp, plugins: Array<Plugin & Ob
   let promiseDepth = 0
 
   async function executePlugin (plugin: Plugin & ObjectPlugin<any>) {
+    // Short-circuit when not rendering error page
+    if (!nuxtApp.payload.error && errors.length) {
+      throw errors[0]
```

Comment:
@danielroe 
Not sure if this is the right way to check if we're rendering an error page, is this what you had in mind?